agent/connect/ca/provider_vault.go

```diff
@@ -381,19 +384,47 @@ func (v *VaultProvider) setupIntermediatePKIPath() error {
 				Config:      mountConfig,
 			})
 			if err != nil {
-				return err
+				return fmt.Errorf("failed to mount intermediate PKI backend: %w", err)
 			}
+			// Required to determine if we should tune the mount
+			// if the VaultProvider is ever reconfigured.
+			v.isConsulMountedIntermediate = true
+
```

Comment:
On an earlier meeting did we decide that this needed a `fallthrough` like the root-generate switch had?

Reply:
When we talked I thought I should be initializing the mount from within `setupIntermediatePKIPath` but realized it wasn't necessary to do that here. So all we need to do is prevent error-ing out here and other codepaths like `generateIntermediateCSR` below will initialize the mount as necessary.

Reply:
So if we do end up mounting the intermediate PKI, there's no other action to be done (unlike `GenerateRoot` which is actually responsible for mounting+initializing+generating).

I thought about making the code paths for root and intermediate more parallel to each other but deferred it to future Chris